implementations/rust/ockam/ockam/src/attestation.rs

```diff
@@ -0,0 +1,170 @@
+use crate::{OckamError, ProfileIdentifier, ProfileVault};
+use ockam_vault_core::{PublicKey, Secret};
+use serde::{Deserialize, Serialize};
+use serde_big_array::big_array;
+
+big_array! { BigArray; }
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct AttestationRequest {
+    requester_profile_id: ProfileIdentifier,
+    nonce: [u8; 16],
+}
+
+impl AttestationRequest {
+    pub fn requester_profile_id(&self) -> &ProfileIdentifier {
+        &self.requester_profile_id
+    }
+    pub fn nonce(&self) -> [u8; 16] {
+        self.nonce
+    }
+}
+
+impl AttestationRequest {
+    fn new(requester_profile_id: ProfileIdentifier, nonce: [u8; 16]) -> Self {
+        AttestationRequest {
+            requester_profile_id,
+            nonce,
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct AttestationResponse {
+    request: AttestationRequest,
+    #[serde(with = "BigArray")]
+    signature: [u8; 64],
+}
+
+impl AttestationResponse {
+    pub fn request(&self) -> &AttestationRequest {
+        &self.request
+    }
+    pub fn signature(&self) -> [u8; 64] {
+        self.signature
+    }
+}
+
+impl AttestationResponse {
+    fn new(request: AttestationRequest, signature: [u8; 64]) -> Self {
+        AttestationResponse { request, signature }
+    }
+}
+
+pub(crate) struct Attestation {}
+
+impl Attestation {
+    pub(crate) fn generate_attestation_request(
+        requester_profile_id: ProfileIdentifier,
+        _vault: &mut dyn ProfileVault,
+    ) -> ockam_core::Result<Vec<u8>> {
+        // FIXME: generate random nonce using vault
```

Comment:
Should we add random trait to vault? Or should we just use whatever we have on this platform

Reply:
@mikelodder7

Reply:
I think we should have the system specify the RNG. IoT usually have to seed this on boot.

Reply:
@mikelodder7 OsRng? thread_rng()?